geopyspark-backend/geotrellis/src/main/scala/geopyspark/geotrellis/io/geotiff/GeoTiffRDD.scala

```diff
@@ -106,10 +108,63 @@ object GeoTiffRDD {
       case S3 =>
         getS3GeoTiffRDD(sc, keyType, uri, S3GeoTiffRDDOptions.setValues(options))
       case _ =>
-        getHadoopGeoTiffRDD(sc, keyType, new Path(path), HadoopGeoTiffRDDOptions.setValues(options))
+        getHadoopGeoTiffRDD(sc, keyType, new Path(uri), HadoopGeoTiffRDDOptions.setValues(options))
     }
   }
 
+  def get_many(
+    sc: SparkContext,
+    keyType: String,
+    paths: java.util.List[String]
+  ): RasterRDD[_] = {
+    val uris = paths.map{ path => new URI(path) }
+
+    uris
+      .map { uri => 
+        uri.getScheme match {
+          case S3 =>
+            getS3GeoTiffRDD(sc, keyType, uri, S3GeoTiffRDDOptions.default)
+          case _ =>
+            getHadoopGeoTiffRDD(sc, keyType, new Path(uri), HadoopGeoTiffRDDOptions.default)
+        }
+      }
+      .reduce{ (r1, r2) => 
+        keyType match {
+          case PROJECTEDEXTENT =>
+            ProjectedRasterRDD(r1.asInstanceOf[ProjectedRasterRDD].rdd.union(r2.asInstanceOf[ProjectedRasterRDD].rdd))
+          case TEMPORALPROJECTEDEXTENT =>
+            TemporalRasterRDD(r1.asInstanceOf[TemporalRasterRDD].rdd.union(r2.asInstanceOf[TemporalRasterRDD].rdd))
+        }
+      }
+  }
+
+  def get_many(
+    sc: SparkContext,
+    keyType: String,
+    paths: java.util.List[String],
+    options: java.util.Map[String, Any]
+  ): RasterRDD[_] = {
```

Comment:
This should be the only function we expose with python wrapper putting the `uri` in a list, if its not and providing empty param list for options. Otherwise this produces too much code to herd that basically does the same thing.